### PR TITLE
fix module names to reflect Jakarta EE origin

### PR DESCRIPTION
**Pull request by rotty3000**
Signed-off-by: Raymond Augé <rotty3000@apache.org>

### CODE REVIEW

**Comment by rotty3000** [comment]
@markt-asf note there are distinct differences with this PR as I'm trying to follow the request you made in the Jakarta mailing list to use .api consistently. Also, in catalina, I was not sure about persistence name. Would it retain the `java.persistence` or follow the new jakarta "_convention_"?

File: res/bnd/catalina.jar.tmp.bnd [hunk 1]
```diff
@@ -55,15 +55,15 @@ Export-Package: \
...
         modules='\
-            jakarta.ejb-api,\
-            jakarta.mail,\
-            jakarta.persistence-api,\
-            jakarta.xml.rpc-api,\
-            java.xml.ws'
+            jakarta.ejb.api,\
+            jakarta.mail.api,\
+            jakarta.persistence.api,\
+            jakarta.xml.rpc.api,\
```
Comment (markt-asf) [inline res/bnd/catalina.jar.tmp.bnd:61]: XML RPC has been dropped in Jakarta EE 9. We'll need to clean that up in Tomcat. I guess this one can stay as is for now.

File: res/bnd/catalina.jar.tmp.bnd [hunk 2]
```diff
@@ -55,15 +55,15 @@ Export-Package: \
...
+            jakarta.xml.ws.api'
```
Comment (markt-asf) [inline res/bnd/catalina.jar.tmp.bnd:62]: Jakarta XMl WS opted for `jakarta.xml.ws`

File: res/bnd/jaspic-api.jar.tmp.bnd [hunk 3]
```diff
@@ -35,9 +35,9 @@ Provide-Capability: \
     Implementation-Vendor=Apache Software Foundation
 
 -jpms-module-info: \
-    jakarta.security.enterprise;\
+    jakarta.security.enterprise.api;\
```
Comment (markt-asf) [inline res/bnd/jaspic-api.jar.tmp.bnd:38]: The most recent RC (2.0.0-RC2) doesn't define a module name so this should be OK for now.

**Comment by rotty3000** [comment]
See updates.

**Comment by markt-asf** [comment]
Heads up. The Jakarta EE JPMS names are currently a moving target. At least one person thinks there is a policy that the JPMS names should not have the ".api" on the end. I am trying to get clarification on the position because as far as I am aware there is no official policy. We have plenty of time so I suggest we wait for Jakarta EE to stabilise and then align our names with theirs.

**Comment by rotty3000** [comment]
I saw :)

**Comment by markt-asf** [comment]
This has been addressed in 10.0.x and 9.0.x. Note there nay be another change for 10.1.x depending on what naming convention Jakarta EE adopts.

**Pull request closed without merging**